e2e/vaultcompat/vaultcompat_test.go

```diff
@@ -0,0 +1,260 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vaultcompat
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/go-set"
+	"github.com/hashicorp/go-version"
+	nomadapi "github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/testutil"
+	vaultapi "github.com/hashicorp/vault/api"
+	"github.com/shoenig/test/must"
+	"github.com/shoenig/test/wait"
+)
+
+const (
+	binDir  = "vault-bins"
+	envGate = "NOMAD_E2E_VAULTCOMPAT"
+)
+
+func TestVaultCompat(t *testing.T) {
+	if os.Getenv(envGate) != "1" {
+		t.Skip(envGate + " is not set; skipping")
+	}
+	t.Run("testVaultVersions", testVaultVersions)
+}
+
+func testVaultVersions(t *testing.T) {
+	versions := scanVaultVersions(t, getMinimumVersion(t))
+	versions.ForEach(func(b build) bool {
+		downloadVaultBuild(t, b)
+		testVaultBuild(t, b)
+		return true
+	})
+}
+
+func testVaultBuild(t *testing.T, b build) {
+	t.Run("vault("+b.Version+")", func(t *testing.T) {
+		vStop, vc := startVault(t, b)
+		defer vStop()
```

Comment:
I can't resist test code golf: you could `t.Cleanup(vlt.Stop)` down in `startVault()` (ditto for `startNomad()`)

Reply:
I was worried if the agents log during shutdown they will cause the test to fail, though I didn't actually try it ...